src/stk500.c

```diff
@@ -1265,6 +1295,27 @@ static int stk500_set_fosc(const PROGRAMMER *pgm, double v) {
 }
 
 
+static int stk500_get_fosc(const PROGRAMMER *pgm, double *v) {
+  unsigned prescale=0, cmatch=0;
+  double fosc = 0;
+  static unsigned ps[] = {
+    1, 8, 32, 64, 128, 256, 1024
+  };
+  int rc;
+
+  if ((rc = stk500_getparm(pgm, Parm_STK_OSC_PSCALE, &prescale)) != 0
+      || (rc = stk500_getparm(pgm, Parm_STK_OSC_CMATCH, &cmatch)) != 0)
+    return rc;
+
+  if (prescale) {
+  fosc = PDATA(pgm)->xtal / ((cmatch + 1) * 2 * ps[prescale - 1]);
```

Comment:
Indentation *looks* wrong, braces not needed? Would `fosc = prescale? ...: ....;` be crisper and clearer?

Reply:
Actually `fosc` isn't needed either? `*v = prescale? ...: ...;` would do.

Reply:
Strange, I will check the indentation, normally gcc warns about such stupid formatting.
But I would stick with the if / else construct (w/o braces, possibly a leftover of my previous more complicated calculation) b/c the ternary is harder to read:
`fosc = prescale ? PDATA(pgm)->xtal / ((cmatch + 1) * 2 * ps[prescale - 1]) : 0;`

Reply:
Pro tip: if one of the two branches is trivial then making that the first option aids readability:
```
  * v = !prescale? 0: /* here comes what it really is */;
```

Reply:
Thx, I've also seen that fosc is not needed and removed it.